test/integration/component/test_affinity_groups_projects.py

```diff
@@ -0,0 +1,1083 @@
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+# 
+#    http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+from marvin.cloudstackTestCase import cloudstackTestCase, unittest
+from marvin.cloudstackAPI import deleteAffinityGroup
+from marvin.lib.utils import (cleanup_resources,
+                        random_gen)
+from marvin.lib.base import (Account,
+                        Project,
+                        ServiceOffering,
+                        VirtualMachine,
+                        AffinityGroup,
+                        Domain)
+from marvin.lib.common import (get_zone,
+                         get_domain,
+                         get_template,
+                         list_hosts,
+                         list_virtual_machines,
+                         wait_for_cleanup)
+from nose.plugins.attrib import attr
+
+class Services:
+    """Test Account Services
+    """
+
+    def __init__(self):
+       self.services = {
+          "domain": {
+             "name": "NonRootDomain"
+          },
+          "domain_admin_account": {
+             "email": "newtest@test.com",
+             "firstname": "Test",
+             "lastname": "User",
+             "username": "doadmintest",
+             "password": "password"
+          },
+          "account": {
+             "email": "newtest@test.com",
+             "firstname": "Test",
+             "lastname": "User",
+             "username": "acc",
+             "password": "password"
+          },
+          "account_not_in_project": {
+             "email": "newtest@test.com",
+             "firstname": "Test",
+             "lastname": "User",
+             "username": "account_not_in_project",
+             "password": "password"
+          },
+          "project": {
+             "name": "Project",
+             "displaytext": "Project"
+          },
+          "project2": {
+             "name": "Project2",
+             "displaytext": "Project2"
+          },
+          "service_offering": {
+             "name": "Tiny Instance",
+             "displaytext": "Tiny Instance",
+             "cpunumber": 1,
+             "cpuspeed": 100,
+             "memory": 64
+          },
+          "ostype": 'CentOS 5.3 (64-bit)',
+          "host_anti_affinity": {
+                "name": "",
+                "type": "host anti-affinity"
+             },
+          "virtual_machine" : {
+             "hypervisor" : "KVM"
```

Comment:
Why do we specify KVM here? It'd be nice if we can run these tests on other hypervisors as well.